### PR TITLE
Issue91: Moving to javaparser 3.0.0-alpha.6

### PR DESCRIPTION
**Pull request by ftomassetti**
This is a working in progress towards #91

### CODE REVIEW

**Comment by ftomassetti** [comment]
This is blocked by https://github.com/javaparser/javaparser/issues/506

File: java-symbol-solver-core/src/main/java/me/tomassetti/symbolsolver/javaparsermodel/JavaParserFacade.java [hunk 1]
```diff
@@ -531,6 +527,14 @@ private Type getTypeConcrete(Node node, boolean solveLambdas) {
         }
     }
 
+    public Type convertToUsageVariableType(VariableDeclarationExpr var) {
+        Type type = JavaParserFacade.get(typeSolver).convertToUsage(var.getElementType(), var);
+        for (int i=0;i<var.getArrayBracketPairsAfterElementType().size();i++) {
+            type = new ArrayType(type);
```
Comment (matozoid) [inline java-symbol-solver-core/src/main/java/me/tomassetti/symbolsolver/javaparsermodel/JavaParserFacade.java]: This code is full of bugs :) VariableDeclarationExpr can contain multiple variables with varying amounts of brackets. There is not one type. You can iterate the variables in it, and getType() on them - you will get the correct types.

File: java-symbol-solver-core/src/main/java/me/tomassetti/symbolsolver/javaparsermodel/declarations/JavaParserSymbolDeclaration.java [hunk 2]
```diff
@@ -129,7 +129,11 @@ public Type getType() {
             VariableDeclarator variableDeclarator = (VariableDeclarator) wrappedNode;
             if (getParentNode(wrappedNode) instanceof VariableDeclarationExpr) {
                 VariableDeclarationExpr variableDeclarationExpr = (VariableDeclarationExpr) getParentNode(variableDeclarator);
-                return JavaParserFacade.get(typeSolver).convert(variableDeclarationExpr.getElementType(), JavaParserFactory.getContext(wrappedNode, typeSolver));
+                Type type = JavaParserFacade.get(typeSolver).convert(variableDeclarationExpr.getElementType(), JavaParserFactory.getContext(wrappedNode, typeSolver));
+                for (int i=0;i<variableDeclarationExpr.getArrayBracketPairsAfterElementType().size();i++) {
+                    type = new ArrayType(type);
+                }
```
Comment (matozoid) [inline java-symbol-solver-core/src/main/java/me/tomassetti/symbolsolver/javaparsermodel/declarations/JavaParserSymbolDeclaration.java]: See last remark! And if you really must do wrapping, check the static methods on ArrayType :)

File: java-symbol-solver-core/src/main/java/me/tomassetti/symbolsolver/javaparsermodel/JavaParserFacade.java (hunk 1 above)
Comment (ftomassetti) [inline java-symbol-solver-core/src/main/java/me/tomassetti/symbolsolver/javaparsermodel/JavaParserFacade.java]: Currently I am rewriteing javaparser-symbol-model, then I plan to heavily refactor all of this code. All the tests should permit to do some refactoring with confidence.

But I will start fixing this :D

Comment (matozoid) [inline java-symbol-solver-core/src/main/java/me/tomassetti/symbolsolver/javaparsermodel/JavaParserFacade.java]: Double-check the last post here: https://github.com/javaparser/javaparser/issues/237
That details all the changes :)
And thanks for supplying feedback!

**Comment by ftomassetti** [comment]
https://github.com/javaparser/javaparser/pull/507 has been merged, once we have a new release of JavaParser we can complete this task

**Comment by matozoid** [comment]
I'll make a release tonight.

**Comment by ftomassetti** [comment]
The other issues has been fixed, I expect all tests to pass with the new release

**Comment by ftomassetti** [comment]
All tests passing. We should review and merge this one. Thanks to @matozoid for releasing the new version of javaparser

**Comment by ftomassetti** [comment]
Ok, @matozoid commented a previous version of this PR. Now I have just moved to alpha-6.

Merging it.

**Pull request merged**